File: pset2/caesar/caesar.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#define UPPER 65
#define LOWER 97

int main(int argc, string argv[])

{
    string plaintext;

    //Ensures that program is executed with one Command Line Argument
    if (argc != 2)
    {
        printf("Usage: ./caesar k\n");
        return 1;
    }

    //Converting string (k) to int
    int k = atoi(argv[1]);

    //Declaring a string variable
    plaintext = get_string("plaintext: ");
    printf("ciphertext: ");

    //Looping through each character
    for (int i = 0, n = strlen(plaintext); i < n; i++)
    {
        //Check if character is an alphabet
        char p = plaintext[i];

        if (isalpha(p))
        {
            if (isupper(p))
            {
                //Converting to alphabetical index (encipher) then back to ASCII (printing)
                printf("%c", (((p + k) - UPPER) % 26) + UPPER);
            }
            else if (islower(p))
            {
                printf("%c", (((p + k) - LOWER) % 26) + LOWER);
            }
        }
        else
        {
            printf("%c", p);
        }
    }
    printf("\n");
    return 0;
}

File: pset6/sentimentals/caesar.py
import sys
from cs50 import get_string

def main():

    # Ensures only one argument is passed in
    if len(sys.argv) != 2:
        print("Usage: ./caesar k")

    # Converting key to int
    key = int(sys.argv[1])

    print("plaintext: ", end = "")
    plaintext = get_string()

    print("ciphertext: ", end = "")

    # Looping through each character in the plaintext
    for c in plaintext:

        # If alphabetic
        if c.isalpha() == True:

            # If upper case
            if c.isupper() == True:
                # Converting to alphabetical index and back to ASCII
                upper = chr((ord(c) + key - 65) % 26 + 65)
                print(upper, end = "")

            # If lower case
            elif c.islower() == True:
                lower = chr((ord(c) + key - 97) % 26 + 97)
                print(lower, end = "")

        else:
            print(c, end = "")

    # Print newline
    print()

if __name__ == "__main__":
    main()